libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SyncState.java

```diff
@@ -200,6 +205,7 @@ public static SyncState fromJSON(JSONObject sync) throws JSONException {
         state.maxTimeStamp = sync.optLong(SYNC_MAX_TIME_STAMP, -1);
 		state.startTime = sync.optLong(SYNC_START_TIME, 0);
 		state.endTime = sync.optLong(SYNC_START_TIME, 0);
+		state.errorJSON = sync.getString(SYNC_ERROR);
```

Comment:
Could you instead do
```java
state.errorJSON = JSONObjectHelper.optString(sync, SYNC_ERROR, "");
```
Sync states created before the code change will be missing that field, and calling getString(...) will throw a JSONException.

Reply:
Resolve in c1f40ddfd48b9c8cfde422d33a880c4d559244e8